File: src/main/js/pl/superczynski/bit.js
function bit(N) {
  return N.toString(2).replace(/^0+|0+$/g, "")
    .split(1)
    .map(function (f) {
      return f.length
    })
    .sort()
    .pop();
}

console.log(bit(1));
console.log(bit(2));
console.log(bit(3));
console.log(bit(334));
console.log(bit(52));

File: src/main/scala/pl/superczynski/bit.scala
package main.scala.pl.superczynski

/**
  * Created by sebastian on 27/05/16.
  */
object bit {

  def toInt(seq: Option[String]) = {
    seq match {
      case Some(s) => s.length()
      case None => 0
    }
  }

  def main(args: Array[String]): Unit = {
    val number: Int = 1024
    val seq = number
      .toBinaryString
      .reverse
      .dropWhile(_ == 0)
      .reverse
      .split("1")
      .toList
      .sorted
      .reverse
      .headOption

    toInt(seq)
  }
}
